intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduFullPluginVersion.kt

```diff
@@ -0,0 +1,72 @@
+package com.jetbrains.plugin.structure.edu
+
+import com.jetbrains.plugin.structure.base.plugin.PluginProblem
+import com.jetbrains.plugin.structure.base.problems.PropertyNotSpecified
+import com.jetbrains.plugin.structure.base.utils.CompatibilityUtils
+import com.jetbrains.plugin.structure.edu.problems.InvalidVersionError
+
+
+data class EduFullPluginVersion(
+  val pluginVersion: EduPluginVersion,
+  val ideVersion: String,
+  val buildNumber: String
+): Comparable<EduFullPluginVersion> {
+  companion object {
+    fun fromString(version: String?): EduFullPluginVersion {
+      if (version.isNullOrEmpty()) fail(version)
+      val parts = version.split("-")
+      if (parts.size != 3 || parts.any { it.isEmpty() }) fail(version)
+      return EduFullPluginVersion(EduPluginVersion.fromString(parts[0]), parts[1], parts[2])
+    }
+
+    fun createIfValid(version: String?, problems: MutableList<PluginProblem>): EduFullPluginVersion? {
+      if (version.isNullOrBlank()) {
+        problems.add(PropertyNotSpecified(VERSION))
+        return null
+      }
+
+      return try {
+        fromString(version)
+      } catch (e: IllegalArgumentException) {
+        problems.add(InvalidVersionError(version))
+        null
+      }
+    }
+
+    private fun fail(version: String?): Nothing {
+      throw IllegalArgumentException("$version doesn't represent full edu plugin version")
+    }
+  }
+
+  fun asLong() = CompatibilityUtils.versionAsLong(pluginVersion.x, pluginVersion.y, pluginVersion.z)
+
+  override fun compareTo(other: EduFullPluginVersion) = pluginVersion.compareTo(other.pluginVersion)
+}
+
+data class EduPluginVersion(val x: Int, val y: Int, val z: Int = 0) : Comparable<EduPluginVersion> {
+  companion object {
+    fun fromString(version: String): EduPluginVersion {
+      if (version.isEmpty()) throw IllegalArgumentException("Edu plugin version shouldn't be empty")
+
+      val components = version.split(".")
+      if (components.size < 2 || components.any { it.toIntOrNull() == null }) {
+        throw IllegalArgumentException("Edu plugin version is invalid: $version")
+      }
+
+      return EduPluginVersion(
+        components[0].toInt(),
+        components[1].toInt(),
+        components.getOrNull(2)?.toInt() ?: 0
+      )
+    }
+  }
+  override fun compareTo(other: EduPluginVersion) = when {
```

Comment:
I've seen almost the same comparator in KtorVersion

Reply:
Yes, it was the same in ktor version. But since we have no base class here, there is no good place to store such a thing.

Reply:
`structure-base` module. Let's create VersionsUtils